src/libraries/System.Net.Quic/tests/FunctionalTests/MsQuicTests.cs

```diff
@@ -1082,5 +1063,42 @@ public async Task Read_ReadsCompleted_ReportedBeforeReturning0()
                     Assert.True(serverStream.ReadsClosed.IsCompletedSuccessfully);
                 });
         }
+
+        [Fact]
+        [OuterLoop("May take several seconds")]
+        public async Task IdleTimeout_ThrowsQuicException()
+        {
+            QuicListenerOptions listenerOptions = new QuicListenerOptions()
+            {
+                ListenEndPoint = new IPEndPoint(IPAddress.Loopback, 0),
+                ApplicationProtocols = new List<SslApplicationProtocol>() { ApplicationProtocol },
+                ConnectionOptionsCallback = (_, _, _) =>
+                {
+                    var serverOptions = CreateQuicServerOptions();
+                    serverOptions.MaxInboundBidirectionalStreams = 1;
+                    serverOptions.MaxInboundUnidirectionalStreams = 1;
+                    serverOptions.IdleTimeout = TimeSpan.FromSeconds(5);
+                    return ValueTask.FromResult(serverOptions);
+                }
+            };
+            (QuicConnection clientConnection, QuicConnection serverConnection) = await CreateConnectedQuicConnection(null, listenerOptions);
+
+            await using (clientConnection)
+            await using (serverConnection)
+            {
+                using QuicStream clientStream = await clientConnection.OpenOutboundStreamAsync(QuicStreamType.Unidirectional);
+                await clientStream.WriteAsync(new byte[1]);
+                using QuicStream serverStream = await serverConnection.AcceptInboundStreamAsync();
+
+                var acceptTask = serverConnection.AcceptInboundStreamAsync();
+
+                // Wait for idle timeout
+                await Task.Delay(TimeSpan.FromSeconds(10));
```

Comment:
Does it need to be 10 seconds?  Can we decrease both the idle timeout and this value?

Reply:
Can we write the test in a way that's more resilient?  For example, issue a read against the stream, time how long it takes for it to fail, and then validate the time is within some acceptable range?

Reply:
I don't see how that makes the test more resilient... The way I see it we care about the `QuicErrorCode` thrown in the `QuicException`

Reply:
It's a combination of resiliency and how much time we're willing to spend in CI.  The way you have it written, you're forced to choose a large value that makes the test take much longer in hopes that it'll be long enough to catch any flakiness in CI.  The way I suggested, you don't have to choose a time to wait at all; you simply wait for the read to complete, validate it threw the expected exception, _and_ validate that the time it took to complete was within some acceptable range.

Reply:
>  I'm not sure what purpose this Task.Delay is serving

I noted above that the delay can be removed, the test can wait/block on the asserts that follow so that there is no unnecessary delay. My question was more about how long do you think the `IdleTimeout` value should be.

Reply:
Ok.  I think it should be short, like 1 second, unless there's something about a short timeout that makes that problematic.